packages/gatsby-theme/src/components/Button/index.js

```diff
@@ -39,12 +39,12 @@ const Button = forwardRef(({ to, variant, as, children, ...props }, ref) => {
 Button.propTypes = {
   to: PropTypes.string,
   variant: PropTypes.string,
-  as: PropTypes.oneOfType([PropTypes.string, PropTypes.elementType]),
-  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+  as: PropTypes.node,
+  children: PropTypes.node.isRequired,
```

Comment:
For the `as` prop, I think this is correct as is. PropTypes.node represents something that can be directly rendered, like when passing a component to a children. In the case of this prop, we can pass a string (for example, `as="div"`) or a component (`as={MyComponent}`). 

We should check to see what `PropTypes.elementType` to see if we can leave it.